tests/unit-tests/gateways/paypal/request.php

```diff
@@ -0,0 +1,261 @@
+			'invoice',
+			'item_name_1',
+			'item_number_1',
+		);
+
+		$query_string = wp_parse_url( $request_url, PHP_URL_QUERY )
+			? wp_parse_url( $request_url, PHP_URL_QUERY )
+			: '';
+		$query_array = array();
+		parse_str( $query_string, $query_array );
+		foreach ( $fields_limited_to_127_chars as $field_name ) {
+			$this->assertLessThanOrEqual( 127, strlen( $query_array[ $field_name ] ) );
+		}
+
+		// Check that there is actually only one item for order with URL length > limit.
+		$this->assertFalse( array_key_exists( 'item_name_2', $query_array ) );
+
+		$this->check_shipping_tax( $query_array, $shipping_tax_included );
+
+		// Remove order and created products.
+		$this->clean_up();
+
+	}
+
+	/**
+	 * Test small order with fewer items, URL length should be < 2083 characters.
+	 *
+	 * @param int  $product_count         Number of products to include in the order.
+	 * @param bool $shipping_tax_included Whether the shipping tax should be included or not.
+	 * @param bool $testmode              Whether to use Paypal sandbox.
+	 *
+	 * @throws WC_Data_Exception
+	 */
+	protected function check_small_order( $product_count, $shipping_tax_included, $testmode ) {
+		$request_url = $this->get_request_url( $product_count, $testmode );
+
+		$this->check_order_common_props( $request_url, $testmode );
+
+		$query_string = wp_parse_url( $request_url, PHP_URL_QUERY )
+			? wp_parse_url( $request_url, PHP_URL_QUERY )
+			: '';
+		$query_array = array();
+		parse_str( $query_string, $query_array );
+
+		// Check that there are $product_count line items in the request URL.
+		// However, if shipping tax is included, there is only one item.
+		if ( $shipping_tax_included ) {
+			$product_count = 1;
+		}
+
+		for ( $i = 1; $i <= $product_count; $i++ ) {
+			$this->assertTrue( array_key_exists( 'item_name_' . $i, $query_array ), 'Item name ' . $i . ' does not exist' );
+			$this->assertTrue( array_key_exists( 'quantity_' . $i, $query_array ) );
+			$this->assertTrue( array_key_exists( 'amount_' . $i, $query_array ) );
+			$this->assertTrue( array_key_exists( 'item_number_' . $i, $query_array ) );
+
+			$this->assertLessThanOrEqual( 127, strlen( $query_array[ 'item_name_' . $i ] ) );
+			$this->assertLessThanOrEqual( 127, strlen( $query_array[ 'item_number_' . $i ] ) );
+
+		}
+
+		$this->check_shipping_tax( $query_array, $shipping_tax_included );
+
+		// Remove order and created products.
+		$this->clean_up();
+	}
+
+	/**
+	 * @throws WC_Data_Exception
+	 */
+	public function test_request_url() {
+		// User set up.
+		$this->user = $this->factory->user->create( array(
+			'role' => 'administrator',
+		) );
+		wp_set_current_user( $this->user );
+
+		// Paths through code changed by:
+		// $sandbox (true/false)
+		// wc_tax_enabled() === get_option( 'woocommerce_calc_taxes' ) === 'yes'
+		// wc_prices_include_tax() === wc_tax_enabled() && 'yes' === get_option( 'woocommerce_prices_include_tax' );.
+		$correct_options = array(
+			[ 'no', 'no', false ],
```

Comment:
This syntax breaks under PHP 5.3/5.2 - Travis had some errors. Can you replace with standard `array` syntax?